plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -653,6 +655,24 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         return Optional.of(new JdbcTypeHandle(Types.NUMERIC, Optional.of("decimal"), Optional.of(decimalType.getPrecision()), Optional.of(decimalType.getScale()), Optional.empty(), Optional.empty()));
     }
 
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<SortItem> sortOrder)
+    {
+        Map<String, JdbcColumnHandle> columns = getColumns(session, handle).stream()
+                .collect(toImmutableMap(JdbcColumnHandle::getColumnName, identity()));
+
+        // PostgreSQL by default orders lowercase letters before uppercase, which is different from Trino
+        // TODO We could still push the sort down if we could inject a PostgreSQL-specific syntax for selecting a collation for given comparison.
```

Comment:
Actually this may be simple to do for PostgreSQL, see https://www.postgresql.org/docs/13/collation.html

```
ORDER BY a COLLATE ...
```

Do you want to give it a try?

Otherwise, I'm fine with the change as is.

Reply:
> Actually this may be simple to do for PostgreSQL,

i should say: simple to do for TopN pushdown, as it's handled in the JdbcClient class
Adding COLLATE for comparisons would be a QueryBuilder change, so less simple.

Reply:
I copied the comment from the predicate pushdown - but I agree it looks like an easy fix.

Let me give it a shot.